Dockerfile

```diff
@@ -0,0 +1,25 @@
+ARG ROS_DISTRO=galactic
+
+FROM ros:$ROS_DISTRO-ros-base
+
+# Use bash instead of sh
+SHELL ["/bin/bash", "-c"]
+
+WORKDIR /ros_ws
+
+# Update Ubuntu Software repository
+RUN apt update && apt upgrade -y && apt install -y \
+        ros-$ROS_DISTRO-joy-linux && \
+    apt clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create and initialise ROS workspace
+
+COPY ./joy2twist ./src/joy2twist
+
+RUN mkdir build && \
+    source /opt/ros/$ROS_DISTRO/setup.bash && \
+    colcon build
```

Comment:
All above steps could be reduced to `COPY` and then singe `RUN` involving `apt update`, using `rosdep` instead of `apt install` and then finally `colcon build` and cleaning `apt cache`.

Reply:
:heavy_check_mark: